### PR TITLE
ibdiags: Use cl_qmap instead of glib hashtable

### PR DESCRIPTION
**Pull request by jgunthorpe**
These do basically the same thing for this purpose, no reason to have a
glib dependency for such a simple data structure.

Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>

### CODE REVIEW

File: libibnetdisc/src/ibnetdisc.c [hunk 1]
```diff
@@ -698,7 +709,12 @@ void add_to_portlid_hash(ibnd_port_t * port, GHashTable *htable)
 		/* We add the port for all lids
 		 * so it is easier to find any "random" lid specified */
 		for (lid = base_lid; lid <= (base_lid + lid_mask); lid++) {
-			g_hash_table_insert(htable, GINT_TO_POINTER(lid), port);
+			struct lid2guid_item *item;
+
+			item = malloc(sizeof(*item));
+			if (item)
+				cl_qmap_insert(&f_int->lid2guid, lid,
```
Comment (weiny2) [inline libibnetdisc/src/ibnetdisc.c:717]: I believe you need to set item->port = port before the insert.

I'm also not a huge fan of having to allocate here...  But there is nothing we can do about it other than having an internal ibnd_port_t object which has the cl_map_item_t in it.  Perhaps we should just make a comment that we don't think the alloc will be enough of a problem to justify the complexity of an other 'internal' data structure.

File: libibnetdisc/src/ibnetdisc.c [hunk 2]
```diff
@@ -698,7 +709,12 @@ void add_to_portlid_hash(ibnd_port_t * port, GHashTable *htable)
 		/* We add the port for all lids
 		 * so it is easier to find any "random" lid specified */
 		for (lid = base_lid; lid <= (base_lid + lid_mask); lid++) {
-			g_hash_table_insert(htable, GINT_TO_POINTER(lid), port);
+			struct lid2guid_item *item;
+
+			item = malloc(sizeof(*item));
```
Comment (weiny2) [inline libibnetdisc/src/ibnetdisc.c:714]: calloc?  Or whatever the current rdma-core preferred alloc is...

**Review by weiny2 — CHANGES_REQUESTED** [review]

File: libibnetdisc/src/ibnetdisc.c (hunk 2 above)
Comment (jgunthorpe) [reply libibnetdisc/src/ibnetdisc.c:714]: all members should be initialized..

File: libibnetdisc/src/ibnetdisc.c (hunk 1 above)
Comment (jgunthorpe) [reply libibnetdisc/src/ibnetdisc.c:717]: oops, yes that assignment is needed

**Pull request merged**